src/clinical/clinical-service.ts

```diff
@@ -219,27 +225,46 @@ export const getPaginatedClinicalData = async (
   query: PaginatedClinicalQuery,
 ) => {
   if (!programId) throw new Error('Missing programId!');
-  const start = new Date().getTime() / 1000;
 
   const allSchemasWithFields = await dictionaryManager.instance().getSchemasWithFields();
   // Get all donors + records for given entity
   const { donors, totalDonors } = await donorDao.findByPaginatedProgramId(programId, query);
 
   const taskToRun = WorkerTasks.ExtractEntityDataFromDonors;
-  const taskArgs = [donors as Donor[], totalDonors, allSchemasWithFields, query];
+  const taskArgs = [donors as Donor[], totalDonors, allSchemasWithFields, query.entityTypes, query];
 
   // Return paginated data
   const data = await runTaskInWorkerThread<{ clinicalEntities: ClinicalEntityData[] }>(
     taskToRun,
     taskArgs,
   );
 
-  const end = new Date().getTime() / 1000;
-  L.debug(`getPaginatedClinicalData took ${end - start}s`);
-
   return data;
 };
 
+export const getDonorEntityData = async (donorIds: number[]) => {
+  const allSchemasWithFields = await dictionaryManager.instance().getSchemasWithFields();
+  // Get all donors + records for given entity
+  const donors = await getDonorsByIds(donorIds);
+  const totalDonors = donors.length;
+
+  const paginationQuery: PaginationQuery = {
+    page: 0,
+    sort: 'donorId',
+    pageSize: totalDonors,
+  };
+
+  const taskToRun = WorkerTasks.ExtractEntityDataFromDonors;
+  const taskArgs = [donors, donors.length, allSchemasWithFields, allEntityNames, paginationQuery];
+
+  // Return paginated data
+  const data = await runTaskInWorkerThread<{ clinicalEntities: ClinicalEntityData[] }>(
+    taskToRun,
+    taskArgs,
+  );
+  return data.clinicalEntities;
```

Comment:
Out of habit, I always separate the functions' `return`, so it's easier to discern it from the rest of the logic.
```suggestion

  return data.clinicalEntities;
```